### DISCUSSION TITLE
Please share your experiences. How are you using this project?

### DISCUSSION THREAD

**Discussion by ehfd** [selkies-project/selkies#54] [Show and tell]
We expect many users to use this project for Cloud Gaming, XR, Autonomous Driving, Robotics, and many other advanced technology development projects.

You will help us greatly if you could share the applications of which this project is being used, however detailed you may explain.
This will also provide new ideas across different fields to the community for incubating interdisciplinary research.
Please also write your laboratory or corporation if possible, and the list of publications or results that used/was influenced by this project.

**Comment by ehfd** [comment]
https://is.muni.cz/th/psva5/Bachelor_s_Thesis__Juraj_Kulich.pdf

**Comment by ehfd** [comment]
https://www.youtube.com/watch?v=n_wUMooWhuY

https://www.youtube.com/watch?v=zagBgED7JOI

**Comment by ehfd** [comment]
https://twitter.com/Tiryoh/status/1515592440937590785

**Comment by ehfd** [comment]
https://repositorio-aberto.up.pt/bitstream/10216/135828/2/489863.pdf

**Comment by Ethorbit** [comment]
I'm using this project for high performance / general purpose container desktop streaming, I'm moving my common desktop tasks (browsing, video & photo editing, etc) to individual containers for greater separation and security without hurting graphical performance or usability

I've ported [docker-nvidia-egl-desktop](https://github.com/selkies-project/docker-nvidia-egl-desktop) to [this NixOS config](https://github.com/Ethorbit/nixos-configs/tree/master/nixos/components/desktop-streaming/profiles/selkies-gstreamer) and now I'm streaming several NixOS systems.

**Reply by ehfd** [reply]
Great, thank you! I have heard great things about Nix as well.

**Comment by skipperro** [comment]
I'm currently experimenting with selkies for two reasons, personal and professional.

Personally I really miss [Google Stadia](https://stadia.google.com/gg/) and want to make an open-source replacement for it that you can host on your own GPU servers. In my house I have server-room in basement filled with GPU machines and each family member have only basic PCs or tablets capable of Internet-Browsing. For Gaming we are using Steam Link and it's OK, but I would like something more flexible and not tied to Steam, so that I can really own it. Also everything is already on Linux and games are running through wine/proton, so selkies-gstreamer fits here nicely.

Professionally I'm working for a German company called [Arcware](www.arcware.com) and we already provide PixelStreaming services from our own GPU servers, but right now we are limited to Unreal Engine applications built for Windows. I would like to explore if selkies-gstreamer would be a tool that could expand our product line to support streaming of any application, game or entire OS. Beside the source of the streaming, we are using very similar tech-stack - Signalling, STUN/TURN, WebRTC, NvEnc, etc. In terms of web frontend-support in many areas we are already ahead of what selkies can offer. I guess we could learn much from each other.

This is the best open-source streaming solution that supports web from many that I've tested, but I'm still learning about different ways of setting it up and managing.
@ehfd - If you would find time to offer some support for making custom application-streaming containers or want to get some tips on making improvements to the web-side of your software, feel free to contact me at any time :)

**Reply by ehfd** [reply]
@Skipperro Hi!

This is wonderful news and I am very keen to talk with you.
But I am currently on hiatus from the project because of other things to make me busy.
Estimated time I'll be back on full scale is around next July.

Our collaborators (click on the Discord logo on our main page) should be able to talk to you meanwhile.

While this project was initially designed to work in high-performance computing Kubernetes environments, it naturally allowed it to work on cloud marketplace platforms based on unprivileged Kubernetes containers such as Vast.ai and RunPod, drastically scaling down instance costs. I am very excited to hear people expanding on this opportunity.

**Comment by jamesah2008** [comment]
Can I please get a discord invite? Looks like the original link is down.

**Reply by ehfd** [reply]
https://discord.gg/wDNGDeSW5F

Does this not work? I just checked this one.

**Comment by jamesah2008** [comment]
This one works!

James Hastings Jr

**Reply by ehfd** [reply]
Where did you find an expired link?

**Comment by to-a11y** [comment]
**Chromium file bridge for Selkies**

I use Selkies as a remote Chromium browser.

One reason is that some websites are available from one network or region but not from another. With Selkies I can run Chromium where the required network access exists and use it from almost any workstation through a normal browser.

The main advantage is simplicity: no client software, drivers, browser extensions or full remote desktop are required. Selkies is fast enough that the remote browser feels close to a local one.

The only missing part for me was file handling. I added a small CDP bridge:

remote <input type="file"> opens the local file picker;
selected files are sent directly to the remote Chromium input;
remote downloads are forwarded back to the local browser;
multiple files, iframes, new tabs and Unicode filenames are supported.

Repository:
[https://github.com/to-a11y/selkies-chromium-bridge](https://github.com/to-a11y/selkies-chromium-bridge?utm_source=chatgpt.com)

Demo:
https://github.com/user-attachments/assets/29881aa0-1274-4a53-a6b1-a3f6efed959a

I expected something similar to already exist, because the idea is quite simple and avoids the complexity of a full remote desktop. If Selkies or SealSkin already has a similar solution, I would be interested to compare them.

**Reply by ehfd** [reply]
Interesting, can you leave this detail in https://github.com/selkies-project/selkies/issues/76 so that we can track this in the future?

**Comment by danoki-jm** [comment]
We’re currently testing Selkies in a cutting-edge scientific HPC environment with Open OnDemand, Slurm and Apptainer.
So far, Selkies + PixelFlux looks very promising for remote scientific desktops, visualization and 3D applications, especially with GPU rendering, NVENC and Websocket/WebRTC.
One of the things we find most interesting is the possibility of giving scientists a workstation-like remote experience backed by HPC and supercomputing resources, directly from their browser.

**Reply by ehfd** [reply]
This aligns perfectly with my drive for the project, and I had recently contributed to Apptainer as part of making Selkies work more smoothly as well. I started maintaining Selkies because noVNC and Guacamole couldn't fulfill my needs.

**Reply by danoki-jm** [reply]
That's great to hear. Thank you for all the work you're putting into Selkies and also Apptainer, it really shows. Selkies is one of the most promising solutions we've found for bringing graphical and GPU-accelerated scientific desktops to HPC users.

**Comment by rbakker** [comment]
Working on a headless server with GPU used for computing tasks, it was very frustrating that advanced 3d viewers that are crucial to getting insights in the data could not use that GPU. Until I discovered this project! Only thing I didn't get to work is drag/drop on a web-based app, Selkies seems to catch the file before it reaches the app.
 
It took half a day to actually stumble upon Selkies. I used AI to find a solution for the above and it took two accidental turns before I landed here. First turn was looking for a proper docker image to install vlgrun/turbovnc on. The selkies image seemed to fulfill the requirement, but I thought selkies was just someone's username on github. Then trying to get the docker image running with Apptainer turned out to be very hard, and I looked for documentation myself on the selkies website. Only then I discovered that selkies is a pip-installable service that needs no TurboVNC client and was developed exactly for my use case. Thanks a lot!
